agent/src/beerocks/slave/link_metrics/ieee802_3_link_metrics_collector.cpp

```diff
@@ -0,0 +1,279 @@
+            rtnl_reply.msg_iov     = &io;
+            rtnl_reply.msg_iovlen  = 1;
+            rtnl_reply.msg_name    = &kernel;
+            rtnl_reply.msg_namelen = sizeof(kernel);
+
+            /**
+             * Read as much data as fits in the receive buffer
+             */
+            if ((length = recvmsg(fd, &rtnl_reply, 0)) != 0) {
+                for (msg_ptr = (struct nlmsghdr *)reply; NLMSG_OK(msg_ptr, length);
+                     msg_ptr = NLMSG_NEXT(msg_ptr, length)) {
+                    switch (msg_ptr->nlmsg_type) {
+                    case NLMSG_DONE:
+                        /**
+                         * This is the special meaning NLMSG_DONE message we asked for by using NLM_F_DUMP flag
+                         */
+                        msg_done = 1;
+                        break;
+                    case RTM_NEWLINK:
+                        /**
+                         * This is a RTM_NEWLINK message, which contains lots of information about a link
+                         */
+                        if (get_link_metrics(msg_ptr, local_interface_name, link_metrics)) {
+                            msg_done = 1;
+                            result   = true;
+                        }
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    return result;
+}
+
+/**
+ * @brief Gets link metrics of an Ethernet network interface.
+ *
+ * Gets link metrics for given Ethernet network by means of a Netlink socket using NETLINK_ROUTE
+ * protocol.
+ *
+ * @param[in] local_interface_name Name of the Ethernet network interface.
+ * @param[in, out] link_metris Link metrics structure with read values.
+ *
+ * @return True on success and false otherwise.
+ */
+static bool get_link_metrics(const std::string &local_interface_name, sLinkMetrics &link_metrics)
+{
+    bool result = false;
+
+    /**
+     * Create Netlink socket for kernel/user-space communication.
+     * No need to call bind() as packets are sent only between the kernel and the originating
+     * process (no multicasting).
+     */
+    int fd = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+    if (fd < 0) {
+        LOG(ERROR) << "Failed creating Netlink socket: " << strerror(errno);
+    } else {
+        /**
+         * Get link metrics using Netlink socket
+         */
+        result = get_link_metrics(fd, local_interface_name, link_metrics);
+
+        /**
+         * Clean up and finish properly
+         */
+        close(fd);
+    }
+
+    return result;
+}
+
+ieee802_3_link_metrics_collector::~ieee802_3_link_metrics_collector() {}
+
+bool ieee802_3_link_metrics_collector::get_link_metrics(const std::string &local_interface_name,
+                                                        const sMacAddr &neighbor_interface_address,
+                                                        sLinkMetrics &link_metrics)
+{
+    (void)neighbor_interface_address;
```

Comment:
We discussed this before - we should add an `UNUSED` macro instead (nitpick)

Reply:
Fixed. 
However, I've used the C++11 attribute syntax [[gnu::unused]] instead of __attribute__((unused))
This way we do not have to define the UNUSED in some common header file and include such header.

Reply:
Would be nice to add that to CODING_STYLE.md.